tensorbay/cli/tag.py

```diff
@@ -24,7 +24,16 @@ def _implement_tag(obj: Dict[str, str], tbrn: str) -> None:
     gas = get_gas(**obj)
     dataset_client = get_dataset_client(gas, info)
 
-    _list_tags(dataset_client)
+    if tag_name:
+        _create_tag(dataset_client, info, tag_name)
+    else:
+        _list_tags(dataset_client)
+
+
+def _create_tag(dataset_client: DatasetClientType, info: TBRN, tag_name: str) -> None:
+    dataset_client.create_tag(name=tag_name)
```

Comment:
To create a tag, the `info` cannot be a draft and the dataset should have commit history.
And creating a tag based on a specific revision is supported, the revision of the `info` should be passed to `create_tag()`.